extensions/wikia/WikiaInYourLang/modules/ext.wikiaInYourLang.js

```diff
@@ -155,6 +160,43 @@ require(
 			tracker.track(trackingParams);
 		}
 
+		function getWIYLRequestSentKey() {
+			return 'wikiaInYourLangRequestSent' + retrieveLinkTitle() + cacheVersion;
+		}
+
+		function getWIYLNotificationShownKey() {
+			return 'wikiaInYourLangNotificationShown' + retrieveLinkTitle() + cacheVersion;
+		}
+
+		function getWIYLMessageKey() {
+			return targetLanguage + 'WikiaInYourLangMessage' + retrieveLinkTitle() + cacheVersion;
+		}
+
+		function getWIYLLinkTitlesKey() {
+			return targetLanguage + 'WikiaInYourLangLinkTitles' + cacheVersion;
+		}
+
+		function saveLinkTitle(linkAddress) {
+			var linkAddressAry = linkAddress.split('/'),
+				listOfCachedTitles = '',
+				linkTitle = linkAddressAry[linkAddressAry.length - 1].length === 0 ? 'main' : linkAddressAry[linkAddressAry.length - 1];
+
+			listOfCachedTitles = cache.get(getWIYLLinkTitlesKey());
+			if ( listOfCachedTitles ) {
+				listOfCachedTitles[articleTitle] = linkTitle;
+			} else {
+				listOfCachedTitles = {};
+				listOfCachedTitles[articleTitle] = linkTitle;
+			}
+
+			cache.set(getWIYLLinkTitlesKey(),listOfCachedTitles, cache.CACHE_LONG);
+		}
+
+		function retrieveLinkTitle() {
```

Comment:
1. some excerpt of what you wrote could end up in JSDoc
2. I know that you don't have to, but the smaller the scope of a variable the easier it is to reason when it gets changed and harder to introduce bugs. There are 172 lines of code between declaration and use. It's just easy to forget where this variable is coming from.